common/config/environment.go

```diff
@@ -72,22 +72,34 @@ func (env *Environment) UpdateExternalConfigMap(externalMap map[string]string) {
 	for k, v := range externalMap {
 		env.externalConfigMap.Store(k, v)
 	}
+	env.externalConfigMap.Range(func(key, value interface{}) bool {
+		if _, ok := externalMap[key.(string)]; !ok {
+			env.externalConfigMap.Delete(key)
```

Comment:
In which case this code will be excuted? 

It looks like that the unit tests do not cover this special case.

Reply:
It will be executed when a key of external config deleted.

Reply:
Units tests already covered, please see the change of test file.

Reply:
Why not clear old map and copy from externalMap once.